docs/RFCS/20200401_alter_column_type.md

```diff
@@ -0,0 +1,87 @@
+- Feature Name: ALTER TABLE ... ALTER COLUMN ... SET DATA TYPE
+- Status: WIP
+- Authors: Richard Cai
+- Cockroach Issues: #9851
+
+# Table of Contents
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [High-level summary](#high-level-summary)
+- [Detailed Explanation](#detailed-explanation)
+- [Problems](#Problems)
+- [Other considerations](#Other-considerations)
+- [Alternatives](#alternatives)
+
+# Summary
+Support changing type of a column. (For nontrivial type conversions using a cast).
+
+# Motivation
+Fairly common use case to change a columns data type, especially in ORMs. Highly requested feature.
+
+# High-level Summary
+The idea for a column conversion requiring a cast is to create a new computed column by applying the cast/function to the original column. For each index that contains that column, create a new index that uses all the columns in i but replaces the old column being converted with the new computed column. Swap the old index with the new index, swap the old column with the new column. Then drop the old columns and indexes.
+
+# Detailed Explanation
+## Converting a column that is not part of an index (simple case)
+If were trying to convert column **c** to type **t**
+1. Create a new hidden computed column c' using c::t'
+2. Enqueue a column swap mutation, for c and c'
```

Comment:
... which performs the following set of actions:

Reply:
Added